src/main/java/org/opensearch/neuralsearch/highlight/SemanticHighlighter.java

```diff
@@ -60,8 +60,18 @@ public HighlightField highlight(FieldHighlightContext fieldContext) {
             return null;
         }
 
+        // The pre- and post- tags are provided by the user or defaulted to <em> and </em>
+        String[] preTags = fieldContext.field.fieldOptions().preTags();
+        String[] postTags = fieldContext.field.fieldOptions().postTags();
```

Comment:
I think we need to check that pre and post tags are actually the same opening and closing tag. E.g. <em>/</em> is valid, and <br>/</em> is probably not valid. Do we have such check?

Reply:
There were similar conversation, we will keep use core for the validation part to keep existing behavior. https://github.com/opensearch-project/neural-search/pull/1254#discussion_r2025748835